src/Jackalope/Transport/DoctrineDBAL/CachedClient.php

```diff
@@ -134,10 +134,23 @@ public function getNode($path)
 
         $cacheKey = "nodes: $path, ".$this->workspaceName;
         if (isset($this->caches['nodes']) && (false !== ($result = $this->caches['nodes']->fetch($cacheKey)))) {
+            if ('' === $result) {
+                throw new ItemNotFoundException("Item $path not found in workspace ".$this->workspaceName);
+            }
+
             return $result;
         }
 
-        $node = parent::getNode($path);
+        try {
+            $node = parent::getNode($path);
+        } catch (\Exception $e) {
+            if (isset($this->caches['nodes'])) {
+                $this->caches['nodes']->save($cacheKey, '');
```

Comment:
Would it make more sense to make this a meaningful value? e.g. `__no_item__` and use a constant?